R/create_query_data.R

```diff
@@ -0,0 +1,894 @@
+        " or a list of data frames and `smq_select` objects.\n",
+        "An object of the following class was provided: ",
+        class(values$definition)
+      )
+    )
+  }
+  obj
+}
+
+#' Asserts Requirements for Terms for Queries
+#'
+#' The function checks the requirements for terms for queries provided by the
+#' user. The terms could have been provided directly in the query definition or
+#' via a user provided function for accessing a SMQ or SDG database.
+#'
+#' @param terms Terms provided by user
+#'
+#' @param expect_query_name Is the `QUERY_NAME` column expected?
+#'
+#' @param expect_query_id Is the `QUERY_ID` column expected?
+#'
+#' @param source_text Text describing the source of the terms, e.g., `"the data
+#'   frame provided for the `definition` element"`.
+#'
+#' @return An error is issued if
+#'
+#' - `terms` is not a data frame,
+#' - `terms` has zero observations,
+#' - the `TERM_LEVEL` variable is not in `terms`,
+#' - neither the `TERM_NAME` nor the `TERM_ID` variable is in `terms`,
+#' - `expect_query_name == TRUE` and the `QUERY_NAME` variable is not in `terms`,
+#' - `expect_query_id == TRUE` and the `QUERY_ID` variable is not in `terms`,
+#'
+#' @examples
+#'
+#' try(
+#'   assert_terms(
+#'     terms = 42,
+#'     source_text = "object provided by the `definition` element")
+#' )
+#'
+#' @export
+#'
+#' @seealso [create_query_data()], [query()]
+#'
+#' @keywords assertion
+#'
+#' @author Stefan Bundfuss
+assert_terms <- function(terms,
+                         expect_query_name = FALSE,
+                         expect_query_id = FALSE,
+                         source_text) {
+  if (!is.data.frame(terms)) {
+    abort(paste0(source_text,
+                 " is not a data frame but ",
+                 what_is_it(terms),
+                 "."))
+  }
+
+  if (nrow(terms) == 0) {
+    abort(paste0(
+          source_text,
+          " does not contain any observations."))
+  }
+
+  vars <- names(terms)
+  if (!"TERM_LEVEL" %in% vars) {
+    abort(
+      paste0("Required variable `TERM_LEVEL` is missing in ",
+            source_text,
+            ".")
+    )
+  }
+  if (expect_query_name) {
+    if (!"QUERY_NAME" %in% vars) {
+      abort(
+        paste0("Required variable `QUERY_NAME` is missing in ",
+               source_text,
+               ".")
+      )
+    }
+  }
+  if (expect_query_id) {
+    if (!"QUERY_ID" %in% vars) {
+      abort(
+        paste0("Required variable `QUERY_ID` is missing in ",
+               source_text,
+               ".")
+      )
+    }
+  }
+  if (!"TERM_NAME" %in% vars & !"TERM_ID" %in% vars) {
+    abort(
+      paste0(
+        "Variable `TERM_NAME` or `TERM_ID` is required.\n",
+        "None of them is in ",
+        source_text,
+        ".\n",
+        "Provided variables: ",
+        enumerate(vars)
+      )
+    )
+  }
+}
+
+#' Create an `smq_select` object
+#'
+#' @param name Name of the query used to select the definition of the query.
```

Comment:
add "from the central databse"?  like in `sdg_select()`?

Reply:
I have added "from the company database" and updated the text for `sdg_select()` as well.